theano/gpuarray/elemwise.py

```diff
@@ -41,6 +41,28 @@ def get_scal(dt):
     return scalar.get_scalar_type(dt)
 
 
+def max_inputs_to_GpuElemwise(node_or_outputs):
+    if isinstance(node_or_outputs, Apply):
+        outputs = node_or_outputs.outputs
+    else:
+        outputs = node_or_outputs
+    ptr_size = 8
+    # We compile code that would do indexing in int64
+    int_size = 8
+
+    # we take the limit from CUDA for now
+    argument_limit = 232
```

Comment:
Why is it so low? The CUDA documentation says we have 4KB, are we really trying to support obsolete GPUs?
4KB would be enough to support > 100 dimensions, instead of choking at 7.